client/utilities/utils.ts

```diff
@@ -6,8 +6,27 @@ export function flattenEquivalent<T>(x: T): T {
 	return x;
 }
 
-export const shuffleArray = (array: unknown[]) =>
-	[...array].sort(() => 0.5 - Math.random());
+/*
+ * implementation of the Fisher–Yates Shuffle algorithm
+ * example here: https://bost.ocks.org/mike/shuffle/
+ */
+export const shuffleArray = (array: unknown[]) => {
+	let currentIndex = array.length;
+
+	// While there remain elements to shuffle...
+	while (currentIndex != 0) {
```

Comment:
as the last step, this will always swap the first element with itself.  Which is not in itself a problem as it's a no-op, but it was interesting.